File: searching/linear_search/linear_search.js
function linear_search(input, ket) {
    for (let i = 0; i < input.length; i++) {
        if( input[i] === key) return i;
    }
    return -1;
}

function linear_search_v2(input, ket) {
    var count = 0;
    for (const num of input) {
        if (num === key) return count;
        count++;
    }
    return -1;
}

const input = [4, 5, 9, 0, 3, 8, 5, 8, 5, 3, 5, 7, 8, 0, 2];
const key = 2;
console.log(`input: ${input}`);
console.log(`key: ${key}`);
console.log(`linear_search: ${linear_search(input, key)}`);
console.log(`linear_search_v2: ${linear_search_v2(input, key)}`);

File: searching/linear_search/linear_search.py
def linear_search(input: list[int], key: int) -> int:

    for i in range(len(input)):
        if input[i] == key:
            return i

    return -1


def linear_search_v2(input: list[int], key: int) -> int:
    count: int = 0
    for num in input:
        if num == key:
            return count
        count += 1

    return -1


if __name__ == "__main__":
    input: list[int] = [8, 11, 3, 94, 0, 32, 5, 7, 9, 1999, 38, 94]
    key: int = 38
    print(f"{input = }")
    print(f"{key = }")
    print(f"{linear_search(input, key) = }")
    print(f"{linear_search_v2(input, key) = }")
